python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,202 @@
+        if not location:
+            return self._default_warehouse_location(database_name, table_name)
+        return location
+
+    def _write_metadata(self, metadata: TableMetadata, io: FileIO, metadate_path: str):
+        ToOutputFile.table_metadata(metadata, io.new_output(metadate_path))
+
+    # tested on pre-existing database
+    def create_table(
+            self,
+            identifier: Union[str, Identifier],
+            schema: Schema,
+            location: Optional[str] = None,
+            partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+            sort_order: SortOrder = UNSORTED_SORT_ORDER,
+            properties: Properties = EMPTY_DICT,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        location = self._resolve_table_location(location, database_name, table_name)
+        # TODO: give it correct path based on java version of glueCatalog
+        metadata_location = f"{location}/metadata/{uuid.uuid4()}.metadata.json"
+        metadata = new_table_metadata(
+            location=location, schema=schema, partition_spec=partition_spec, sort_order=sort_order,
+            properties=properties
+        )
+        io = load_file_io({**self.properties, **properties}, location=location)
+        self._write_metadata(metadata, io, metadata_location)
+        try:
+            create_table_response = self.client.create_table(
+                DatabaseName=database_name,
+                TableInput={
+                    'Name': table_name,
+                    'Description': '',  # To be fixed
+                    'TableType': 'EXTERNAL_TABLE',
+                    'Parameters': self._construct_parameters(metadata_location),
+                }
+            )
+            # TODO: check response
+            load_table_response = self.client.get_table(DatabaseName=database_name, Name=table_name)
+            glue_table = load_table_response['Table']
+        except self.client.exceptions.AlreadyExistsException as e:
+            raise TableAlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+
+        return self._glue_to_iceberg(glue_table, io)
+
+    # tested
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            load_table_response = self.client.get_table(DatabaseName=database_name, Name=table_name)
+            self._check_response(load_table_response)
+        except self.client.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {table_name}") from e
+        # TODO: may need to add table properties to the io too
+        io = load_file_io(
+            {**self.properties, **load_table_response['Table']['Parameters']},
+            load_table_response['Table']['StorageDescriptor']['Location'])
```

Comment:
similar to the comment above, `load_table_response['Table']['StorageDescriptor']['Location']` is not the source of truth. Here location is used to infer FileIO implementation, if we really want we should use the `metadata_location` path.